src/config/aegir-tsconfig.json

```
@@ -0,0 +1,26 @@
+{
+    "compilerOptions": {
+        "allowJs": true,
+        "checkJs": true,
+        "target": "es2018",
+        "module": "CommonJS",
+        "moduleResolution": "node",
+        "resolveJsonModule": true,
+        "stripInternal": true,
+        "strict": false,
+        "alwaysStrict": true,
+        "strictNullChecks": true,
+        "noFallthroughCasesInSwitch": true,
+        "preserveConstEnums": true,
+        "noEmitOnError": true,
```

```suggestion
        "noEmitOnError": false,
```

Shouldn't we generate `*.d.ts` files even though some TypeScript related errors were reported in `*.js` files?

---

I don't think it is a good idea, because chances are docs got out of sync from the implementation. I would personally not generate so we would resolve this kind of issues.